app/classes/content_filter.rb

```diff
@@ -0,0 +1,98 @@
+# filters which are applied in Query
+module ContentFilter
+
+# To add a new Content Filter:
+# * Indicates unnecessary for new observation filter -- it's done automatically.
+#
+#   Add tests, e.g., to TestUserContentFilter
+#   Supplement fixtures as needed by added tests
+#   Add a filter definition below, and add that definition to #filters below
+#   Add a new checkbox_val method in User
+#   In /config/locales/en.txt define text to be displayed before and next to
+#     check boxes, using prefs_obs_filters_has_images: as a model.
+# * Supplement _prefs_filters.html.erb as needed
+# * Add filter to AccountController#prefs_types & #update_content_filter
+# * For Observation filter, supplement Query::Initializers::ObservationFilters
+# * Supplement ApplicationController#show_index_of_objects as needed
+#   To filter another object, create a new initializer and include in
+#     appropriate searches.
+#
+# To be able to override the new filter in Advanced Searches, at least:
+#   Add tests, e.g., to TestAdvancedSearchFilters
+#   Supplement fixtures as needed by added tests
+#   In /config/locales/en.txt define text to be displayed before and next to
+#     radio boxes, using advanced_search_filter_has_images: as a model.
+# * Supplement _advanced_search_filters.html.erb as needed
+# * Supplement ObservationController#advanced_search_form as needed.
+# * Supplement ApplicationController#show_index_of_objects as needed.
+# * Supplement Query::RssLogBase as needed.
+#
+# To also add it as a Pattern Search
+#   Supplement PatternSearchTest
+#   Supplement classes/pattern_search/observation.rb or -- if it's not an
+#     Observation search -- add a new pattern_search file and class.
+#
+# There are probably other steps/files I've forgotten. JDC 2016-09-15
+
+  ### filter definitions ###
+  # In the order you want filters to appear in advanced_search
+  # name:         filter name, as string
+  # sym:          filter mame, as symbol
+  # model:        model on which filter operates.  Used by ContentFilter
+  # checkbox:     prefs form checkbox, e.g., :has_images_checkbox.
+  #               Used by prefs view
+  # checked_val:  value when checkbox checked
+  # off_val:      value when filter is off
+  # on_vals:      array of allowed values when filter is on,
+  #               in order you want them to appear in advanced_filters
```

Comment:
Interesting point. For the moment, I'd like to allow non-binary filters.
I originally envisioned these as non-binary and originally implemented them that way. E.g., a user could persistently choose to exclude non-vouchered Observations, to exclude vouchered Observations.
@AlanRockefeller and @pellaea convinced me that there wasn't a good use case for persistently excluding vouchered (or imaged), so I ratcheted these two back to binary.
But I keep thinking about other non-binary possibilities, e.g., a location filter based on continent, country, etc.